### PR TITLE
Feature/text field

### PR DESCRIPTION
**Pull request by chromaticWaster**
### Description:

Adding of Text field component which expanded into adding a base form class which other input permutations extend. 

The following components were added.

Text field component 
Password field component 
Numeric field component 

Stories are added for the the above components along with a shared class for common input stories.

Icons added

EyeHidden
EyeVisible
LockClosed
LockOpen
Plus
Minus

### Screenshots (if appropriate):

Test results

image

### All Submissions:

* [x] I have followed the guidelines in the Contributing document.
* [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.

### Changes to Core Features:

* [x] I have added an explanation of what my changes do and why I would like to include them.
* [x] I have written new tests for my core changes, as applicable.
* [x] I have successfully ran tests with my changes locally.
* [x] I have added/updated docs, as applicable.

### CODE REVIEW

File: src/core/OmniFormElement.ts [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+            }
+        }
+    }
+
+    _focusLost() {
+        if (this.disabled) {
+            return;
+        }
+
+        if (!this.value) {
```
Comment (BOTLANNER) [inline src/core/OmniFormElement.ts:134]: Also check if _labelElement exist

File: src/core/OmniInputStories.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        args: {
+            label: 'Disabled',
+            disabled: true
+        },
+        play: async (context: StoryContext) => {
+            const input = within(context.canvasElement).getByTestId<T>('test-field');
+
+            //Disabled class test.
+            const disabledAttribute = input.attributes.getNamedItem('disabled');
+            /*const disabledClass = input.classList.contains('disabled');*/
```
Comment (BOTLANNER) [inline src/core/OmniInputStories.ts:173]: No commented code please

**Review by BOTLANNER — CHANGES_REQUESTED** [review]

**Review by BOTLANNER — APPROVED** [review]

File: src/core/OmniFormElement.ts [hunk 3]
```diff
@@ -0,0 +1,375 @@
+import { css, CSSResultGroup, html, LitElement, nothing, TemplateResult } from 'lit';
+export { ifDefined } from 'lit/directives/if-defined.js';
+import { property, query } from 'lit/decorators.js';
+import { ClassInfo, classMap } from 'lit/directives/class-map.js';
+import ComponentStyles from '../styles/ComponentStyles.js';
+
+/**
+ * Base class used by components to share common properties styles and functionality
```
Comment (stefan505) [inline src/core/OmniFormElement.ts:8]: Is the description correct?

File: src/core/OmniFormElement.ts [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+ * @cssprop --omni-form-focussed-border-color - Form focussed border color.
+ * @cssprop --omni-form-focussed-label-color - Form focussed label color.
+ *
+ * @cssprop --omni-form-error-label-color - Form error label color.
+ * @cssprop --omni-form-error-border-color - Form error border color.
+ *
+ * @cssprop --omni-form-label-disabled-color - Form label disabled color.
+ * @cssprop --omni-form-disabled-border-color - Form disabled border color.
+ * @cssprop --omni-form-disabled-background-color - Form disabled background color.
+ * @cssprop --omni-label-disabled-focus-background-color - Form label disabled focus background color.
```
Comment (stefan505) [inline src/core/OmniFormElement.ts:44]: Is this prop name correct?

File: src/core/OmniFormElement.ts [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+ */
+export class OmniFormElement extends LitElement {
+    /**
+     * Text label.
+     * @attr
+     */
+    @property({ type: String, reflect: true }) label: string;
+
+    /**
+     * The value entered into the form.
```
Comment (stefan505) [inline src/core/OmniFormElement.ts:75]: Is this description correct?

File: src/core/OmniFormElement.ts [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+    @property({ type: String, reflect: true }) error: string;
+
+    /**
+     * Indicator if the component should be disabled.
+     * @attr
+     */
+    @property({ type: Boolean, reflect: true }) disabled = false;
+
+    @query('.form-container')
+    private _formContainer: HTMLElement;
```
Comment (stefan505) [inline src/core/OmniFormElement.ts:105]: Should this not be called _formContainerElement?

File: src/core/OmniFormElement.ts [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+                .border {
+                    position: absolute;
+                    top: var(--omni-form-border-top, 0px);
+                    bottom: var(--omni-form-border-bottom, 0px);
+                    left: var(--omni-form-border-left, 0px);
+                    right: var(--omni-form-border-right, 0px);
+
+                    border-width: var(--omni-form-border-width, 1px);
+                    border-radius: var(--omni-form-border-radius, 4px);
+                    border-style: solid;
```
Comment (stefan505) [inline src/core/OmniFormElement.ts:182]: CSS var?

**Review by stefan505 — COMMENTED** [review]
Please check comments

File: src/core/OmniFormElement.ts (hunk 3 above)
Comment (chromaticWaster) [reply src/core/OmniFormElement.ts:8]: Updated description to reference form controls.

File: src/core/OmniFormElement.ts (hunk 4 above)
Comment (chromaticWaster) [reply src/core/OmniFormElement.ts:44]: Updated to be more descriptive

File: src/core/OmniFormElement.ts (hunk 5 above)
Comment (chromaticWaster) [reply src/core/OmniFormElement.ts:75]: Updated description.

File: src/core/OmniFormElement.ts (hunk 6 above)
Comment (chromaticWaster) [reply src/core/OmniFormElement.ts:105]: Updated to suggested name.

File: src/core/OmniFormElement.ts (hunk 7 above)
Comment (chromaticWaster) [reply src/core/OmniFormElement.ts:182]: Updated accordingly

**Review by stefan505 — APPROVED** [review]

**Pull request merged**